packages/data/src/components/use-query-select/index.ts

```diff
@@ -0,0 +1,107 @@
+/**
+ * Internal dependencies
+ */
+import useSelect from '../use-select';
+import { META_SELECTORS } from '../../store';
+import memoize from './memoize';
+
+interface QuerySelectResponse {
+	/** the requested selector return value */
+	data: Object;
+
+	/** is the record still being resolved? Via the `getIsResolving` meta-selector */
+	isResolving: boolean;
+
+	/** was the resolution started? Via the `hasStartedResolution` meta-selector */
+	hasStarted: boolean;
+
+	/** has the resolution finished? Via the `hasFinishedResolution` meta-selector. */
+	hasResolved: boolean;
```

Comment:
As mentioned in https://github.com/WordPress/gutenberg/pull/38522#discussion_r802675337, I'm not sure what's the difference between this and `!!data`? We can discuss this in the original thread though.